### PR TITLE
Specflow agent customization example

### LINKED ISSUE #1: Specflow Agent customization example
Could you add the specflow agent customization examples such as adding tags to the features/scenarios?

### PR DESCRIPTION
**Pull request by abasau**
The pull request will resolve #1. Hooks1.cs contains examples of how to add tags to feature and scenario on runtime, add scenario outline parameters to scenario name on RP, attach screenshot of the desktop at the end of scenario.

### CODE REVIEW

File: src/Example.SpecFlow/Hooks/Hooks1.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                        Level = LogLevel.None,
+                        Time = DateTime.UtcNow,
+                        Text = "Screenshot " + DateTime.Now,
+                        Attach = attach
+                    });
+                }
+            }
+        }
+
+        private static void ReportPortalAddin_BeforeScenarioStarted(object sender, TestItemStartedEventArgs e)
```
Comment (nvborisenko) [inline src/Example.SpecFlow/Hooks/Hooks1.cs:23]: I think we should not use nunit in customizations to show examples. Unit provider might be changed and this example will become incorrect.
Let's change this code to show:
* how to modify description
* how to add custom tags

File: src/Example.SpecFlow/Hooks/Hooks1.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            const string pattern = @"^[^\(]+\((.*)\)$";
+
+            // Adding test case parameters to scenario name
+            if (Regex.IsMatch(nunitTestName, pattern))
+            {
+                var nunitTestParameters = Regex.Match(nunitTestName, pattern).Groups.Cast<Group>().ElementAt(1).Value;
+                e.TestItem.Name += ": " + Regex.Replace(nunitTestParameters, @"\,(null|System\.String\[\])$", "");
+            }
+
+            // Workaround for 1024 characters restriction on the test item description
```
Comment (nvborisenko) [inline src/Example.SpecFlow/Hooks/Hooks1.cs]: All limitations should be handled at agent level.

File: src/Example.SpecFlow/Hooks/Hooks1.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            if (string.IsNullOrEmpty(e.TestItem.Description) == false && e.TestItem.Description.Length > 1024)
+            {
+                e.TestItem.Description = e.TestItem.Description.Substring(0, 1024);
+            }
+
+            // Workaround for 256 characters restriction on the test item name
```
Comment (nvborisenko) [inline src/Example.SpecFlow/Hooks/Hooks1.cs]: Agent should cut test `name` and `description`. I submitted issue #2 to resolve it.

**Review by nvborisenko — CHANGES_REQUESTED** [review]
* don't use nunit
* remove limitations part

**Comment by abasau** [comment]
I introduced the dependency on NUnit in order to add scenario parameters to scenario name. It's not possible to get scenario parameters from SpecFlow and the only way to do it is to get them from unit provider. It will not be possible to implement the functionality without NUnit and it will need to be removed it completely. I consider this functionality quite useful since you will not be able to distinguish scenarios with parameters from each other on Report Portal and auto-analysis will not work.
I would still consider leaving the dependency on NUnit in place just to have an example of how to make scenario names unique on RP.

Please let me know what you think.

The limitation parts will be removed. I would implement them on [client-net](https://github.com/reportportal/client-net) library level (not SpecFlow agent level) since it affects all .NET agents.

**Comment by nvborisenko** [comment]
Reflecting input parameters in test name is useful, but this is not general case. Here we need provide simple example of customization.
Let's share nunit specific stuff as code snippets. Somewhere in documentation we will reflect all useful snippets, but we will not use these snippets in code. Example project should be simple.

And agree with cutting length in client, not in agent.

**Review by nvborisenko — APPROVED** [review]

**Comment by nvborisenko** [comment]
Thanks for understanding.

**Pull request merged**